LEAF_Request_Portal/templates/print_form.tpl

```diff
@@ -686,9 +686,23 @@ function doSubmit(recordID) {
     }
 
     function cancelRequest() {
+        dialog_confirm.setTitle('Confirm');
+        <!--{if $submitted > 0}-->
+        dialog_confirm.setContent(
+            '<img src="dynicons/?img=process-stop.svg&amp;w=48" alt="" style="float: left; padding-right: 24px" /> Are you sure you want to cancel this request?<br style="clear: both" /><br />The following reason will be emailed as a notification:<br /><textarea id="cancel_comment" style="width: 100%" rows=3></textarea>'
+        );
+        dialog_confirm.setRequired(<!--{$recordID|strip_tags|escape}-->, function() {
+            let comment = document.querySelector('#cancel_comment').value;
+            if(comment.trim() == '') {
```

Comment:
As opposed to here where adding the code to give a single return would only muddy things up.  Always exceptions to every "rule".